json-endpoints/class.wpcom-json-api-list-roles-endpoint.php

```diff
@@ -60,26 +86,45 @@ static function role_sort( $a, $b ) {
 
 	// /sites/%s/roles/ -> $blog_id
 	function callback( $path = '', $blog_id = 0 ) {
-
 		$blog_id = $this->api->switch_to_blog_and_validate_user( $this->api->get_blog_id( $blog_id ) );
 		if ( is_wp_error( $blog_id ) ) {
 			return $blog_id;
 		}
 
-		if ( ! current_user_can( 'list_users' ) ) {
-			return new WP_Error( 'unauthorized', 'User cannot view roles for specified site', 403 );
-		}
-
 		$roles = array();
 
-		$wp_roles= new WP_Roles();
-		$role_names = $wp_roles->get_names();
-		$role_keys = array_keys( $role_names );
+		$sal_site = $this->get_platform()->get_site( $blog_id );
+		$wp_roles = $sal_site->get_roles();
+
+		// Check if the site is connected and talks to us on a regular basis
+		$is_connected = $sal_site->is_connected_site();
```

Comment:
`is_connected_site` is not defined on Jetpack anywhere 🤔

Reply:
Where should it be defined?